spyder/app/mainwindow.py

```diff
@@ -3101,7 +3102,7 @@ def _dialog_finished(result_code):
             # Check settings and show dialog
             dlg.show()
             dlg.check_all_settings()
-            dlg.exec_()
+            # dlg.exec_()  # Why?

```

Comment:
@ccordoba12 is there a reason why we need the exec here? The preferences dialog is not really acting in a modal way so removing this would make it much easier to add tests for this.